drivers/pinctrl/pinctrl_nrf.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/pinctrl.h>
+
+#include <hal/nrf_gpio.h>
+
+/**
+ * @brief Configure pin settings.
+ *
+ * @param pin Pin configuration.
+ * @param dir Pin direction.
+ * @param input Pin input buffer connection.
+ */
+static void nrf_pin_configure(pinctrl_soc_pin_t pin, nrf_gpio_pin_dir_t dir,
+			      nrf_gpio_pin_input_t input)
+{
+	/* force input direction and disconnected buffer for low power */
+	if (NRF_GET_LP(pin) == NRF_LP_ENABLE) {
+		dir = NRF_GPIO_PIN_DIR_INPUT;
+		input = NRF_GPIO_PIN_INPUT_DISCONNECT;
+	}
```

Comment:
In practice, it will only be applied if adds `bias-pull-up` or `bias-pull-down` to the sleep pins node, but we can remove that degree of _freedom_ if you prefer. What was the current behavior when enabling `rx-pull-up` property?

Reply:
> In practice, it will only be applied if adds `bias-pull-up` or `bias-pull-down` to the sleep pins node, but we can remove that degree of _freedom_ if you prefer.

Right, I see now. At first I didn't understand it well. No need to change anything here, it is good as it is now. Actually, one may even find it useful to have the possibility to use custom pull configuration for a pin in low power.

> What was the current behavior when enabling `rx-pull-up` property?

The pull up was applied only when the device was active. When the device was suspended, all pins were configured with `nrf_gpio_cfg_default()`, so the default reset configuration was applied (input, buffer disconnected, no pull).